### DISCUSSION TITLE
A simple example for body text filtering or changing?

### DISCUSSION THREAD

**Discussion by haegar33** [quoid/userscripts#760] [Q&A]
This extension looks very promising but my javascript knowledge is just above beginner level. It would be great if more example templates were provided as for example how to simply(?) grab the text from a website and filter words or replace them with other?

**Comment by quoid** [comment]
You can use most of the normal [browser/web api](https://developer.mozilla.org/en-US/docs/Web/API) as you would when writing javascript for a webpage. The syntax for a user script is outlined in the [readme](https://github.com/quoid/userscripts?tab=readme-ov-file#table-of-contents), but there might be better resources elsewhere on the web.

> for example how to simply(?) grab the text from a website and filter words or replace them with other?

You would probably want to think about how you might go about this if you were writing code for a webpage to do this via javascript. That would probably get you most of the way. _What have you tried so far?_


Here's a quick example, if you load this into a userscript and then visit [example.net](http://example.net), you should see a custom `h1`.


```
// ==UserScript==
// @name        haegar33 Example
// @description This userscript changes the inner text in the h1 element.
// @match       *://*.example.net/*
// ==/UserScript==

const textElement = document.querySelector("h1");
textElement.innerText = "Hi haegar33"
```

**Comment by haegar33** [comment]
Thanks, that's a good starting point! Actually I found an example for complete text search and replace template:
```
// Get all text nodes in the document
function findTextNodes() {
//    console.log("--- Entering findTextNodes");
    const walker = document.createTreeWalker(
        document.body,
        NodeFilter.SHOW_TEXT,
        null,
        false
    );

    const textNodes = [];
    let node;
    while (node = walker.nextNode()) {
        textNodes.push(node);
    }
    return textNodes;
}

// Replace text content based on your rules
function modifyContent() {
    console.log("--- Entering modifyContent");
    const textNodes = findTextNodes();
    textNodes.forEach(node => {
        node.textContent = node.textContent.replace('old', 'new');
});
}

// Run when page is loaded
document.addEventListener('DOMContentLoaded', modifyContent);
```

**Reply by ACTCD** [reply]
This usage pretty much ruins the entire point of a `TreeWalker`. 
Read the documentation: 
https://developer.mozilla.org/en-US/docs/Web/API/Document/createTreeWalker
